src/modules/authentication/components/Registration/Registration.tsx

```diff
@@ -1,5 +1,231 @@
+			<img src={profileImg} alt="form profile image" />
+		</div>
+
+		<form onSubmit={handleSubmit(onSubmit)}>
+		<div className="container">
+		<div className={`${styles.rowInputs} row` }>
+				<div className="col-md-6">
+				<div className="input">
+				<label className={styles.formLabel} htmlFor="userName">User Name</label>
+				<input   id='userName' type="text" className={`${styles.formInputs} form-control`} placeholder='Enter your Name' 
+				{...register ( 'userName', {required :'name cannot be empty' , pattern : {
+					value : /^\S+\d$/,
+				  
+					message : 'The user name must end with numbers without spaces'
+				  } ,  maxLength : {
+					value : 8,
+					message : 'maximum 8 characters'
+				  }} ) }
+				/>
+
+		{errors.userName && <span className='text-danger'>{errors?.userName?.message}</span>}
+</div>
+				</div>
+				<div className="col-md-6">
+				<div className="input">
+				<label className={styles.formLabel} htmlFor="email">E-mail</label>
+				<input id='email' type="email" className={`${styles.formInputs} form-control`} placeholder='Enter your E-mail'
+				
+				{...register('email',{
+					required : 'email cannot be empty',
+					pattern : {
+					  value : /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/,
+					  message : 'please enter a valid email'
+					}
+				  })}
+				/>
+						{errors.email && <span className='text-danger'>{errors?.email?.message}</span>}
+
+				</div>
+				</div>
+				<div className="col-md-6">
+				<div className="input">
+				<label className={styles.formLabel} htmlFor="country">Country</label>
+				<input id='country' type="text" className={`${styles.formInputs} form-control`} placeholder='Enter your country' 
+				{...register('country' , {required : "please enter a country name"})}
+/>
+				</div>
+				{errors.country && <span className='text-danger'>{errors?.country?.message}</span>}
+
+				</div>
+				<div className="col-md-6">
+				<div className="input">
+				<label className={styles.formLabel} htmlFor="phone">Phone Number</label>
+				<input id='phone' type="tel" className={`${styles.formInputs} form-control`} placeholder='Enter your phone number' 
+				{...register('phoneNumber' , {required : 'please enter your phone number' , pattern : {
+					value:/^01[0125][0-9]{8}$/,
+					message : 'please enter valid egyptian number'
+				  }})}/>
+				</div>
+				{errors.phoneNumber && <span className='text-danger'>{errors?.phoneNumber?.message}</span>}
+
+				</div>
+				<div className="col-md-6">
+				<div className="input position-relative">
+				  <div className="position-relative">
+				  <label className={styles.formLabel} htmlFor="password">Password</label>
+
+				  <input    id='password' type={!isVisible ? "password" : "text"} className={`${styles.formInputs} form-control`} placeholder='Enter your password ' 
+				
+				{...register('password' , {
+					required : 'password cannot be empty',
+					pattern : { value : /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{6,}$/,
```

Comment:
We can extract validation logic into a separate file. This will make the components easier to follow.

Reply:
done